src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx

```diff
@@ -0,0 +1,211 @@
+        result,
+      );
+    } else {
+      processedResults.set(rowIdentifier, [result]);
+    }
+  });
+  const restructuredResults: SubtestsResults[] = Array.from(
+    processedResults,
+    function ([rowIdentifier, result]) {
+      return {
+        key: rowIdentifier,
+        value: result,
+      };
+    },
+  );
+
+  return restructuredResults;
+}
+
+const cellsConfiguration: CompareResultsTableConfig[] = [
+  {
+    name: 'Subtests',
+    key: 'subtests',
+  },
+  {
+    name: 'Base',
+    key: 'base',
+  },
+  { key: 'comparisonSign' },
+  { name: 'New', key: 'new' },
+  {
+    name: 'Status',
+    disable: true,
+    filter: true,
+    key: 'status',
+    possibleValues: ['No changes', 'Improvement', 'Regression'],
+    matchesFunction: (result: CompareResultsItem, value: string) => {
+      switch (value) {
+        case 'Improvement':
+          return result.is_improvement;
+        case 'Regression':
+          return result.is_regression;
+        default:
+          return !result.is_improvement && !result.is_regression;
+      }
+    },
+  },
+  {
+    name: 'Delta(%)',
+    key: 'delta',
+  },
+  {
+    name: 'Confidence',
+    disable: true,
+    filter: true,
+    key: 'confidence',
+    possibleValues: ['Low', 'Medium', 'High'],
+    matchesFunction: (result: CompareResultsItem, value: string) =>
+      result.confidence_text === value,
+  },
+  { name: 'Total Runs', key: 'runs' },
+  { key: 'buttons' },
+  { key: 'expand' },
+];
+
+function resultMatchesSearchTerm(
+  result: CompareResultsItem,
+  searchTerm: string,
+) {
+  return result.test.includes(searchTerm);
+}
+
+function resultMatchesColumnFilter(
+  result: CompareResultsItem,
+  columnId: string,
+  uncheckedValues: Set<string>,
+): boolean {
+  const cellConfiguration = cellsConfiguration.find(
+    (cell) => cell.key === columnId,
+  );
+  if (!cellConfiguration || !cellConfiguration.filter) {
+    return true;
+  }
+
+  const { matchesFunction } = cellConfiguration;
+  for (const filterValue of uncheckedValues) {
+    if (matchesFunction(result, filterValue)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// This function filters the results array using both the searchTerm and the
+// tableFilters. The tableFilters is a map ColumnID -> Set of values to remove.
+function filterResults(
+  results: CompareResultsItem[],
+  searchTerm: string,
+  tableFilters: Map<string, Set<string>>,
+) {
+  if (!searchTerm && !tableFilters.size) {
+    return results;
+  }
+
+  return results.filter((result) => {
+    if (!resultMatchesSearchTerm(result, searchTerm)) {
+      return false;
+    }
+
+    for (const [columnId, uncheckedValues] of tableFilters) {
+      if (resultMatchesColumnFilter(result, columnId, uncheckedValues)) {
+        return false;
+      }
+    }
+
+    return true;
+  });
+}
+
+type ResultsTableProps = {
+  filteringSearchTerm: string;
+};
+
+function SubtestsResultsTable({ filteringSearchTerm }: ResultsTableProps) {
```

Comment:
Can you please file a JIRA ticket to merge this with the other results table? I think we can extract a common component doing the filtering... but I don't want to hold up this PR for this.

Reply:
https://mozilla-hub.atlassian.net/browse/PCF-489